lib/pure/collections/sets.nim

```diff
@@ -1008,7 +1007,7 @@ when isMainModule and not defined(release):
 
     block toSeqAndString:
       var a = toHashSet([2, 7, 5])
-      var b = initHashSet[int]()
+      var b = initHashSet[int](rightSize(a.len))
```

Comment:
Is this change still needed? Does the example work without it?

Reply:
That change is necessary because `toHashSet` also uses `rightSize` for efficiency (to prevent doubling-up however many times).  If you drop the `assert` you could drop the `rightSize`. If you compare the output sets by membership, not by `string`/`$` which depends upon hash order then you could also drop the `rightSize`.  OTOH, if you are considering it a "good example" then `rightSize` is good style since you start with the right size of an output table which is always good style if you know it (and sadly not often provided by hash table libraries).

Reply:
Oh, also it's part of a `block toSeqAndString` which, at least to me, made it seem more correct to keep the stringification but make the `HashSet` table size the same, not whatever the default initial size is.  "Hash order" is really a concept inherently relative to a given hash table size since it's `hash() and mask`.  So, if you did want to drop the `$`s we would also probably want to change the block title.  But I thin it's fine as-is.

Reply:
The change to `sugar.nim` is, however, not strictly necessary but the consistency might be considered "better style".